### PR TITLE
Llama3.1

### PR DESCRIPTION
**Pull request by mylesgoose**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Screenshots
Please attach before and after screenshots of the change if applicable.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by mylesgoose** [comment]
this is just the llama 3.1 models i removed the mllama 3.1 code and opened a new mllama branch for that. I have not run a test on the code

**Comment by mylesgoose** [comment]
the code check that is falling apparently can be fixed with this  format the transformer_lens/loading_from_pretrained.py file using black, follow these steps:

Clone the repository (if you haven't already):

git clone https://github.com/TransformerLensOrg/TransformerLens.git
cd TransformerLens
Install black (if it's not already installed):

pip install black
Run black on the specific file:

black transformer_lens/loading_from_pretrained.py
Commit the changes:

git add transformer_lens/loading_from_pretrained.py
git commit -m "Format transformer_lens/loading_from_pretrained.py using black"
git push origin <your-branch-name>
These steps will format the file and push the changes to your repository.

**Comment by mylesgoose** [comment]
black transformer_lens/loading_from_pretrained.py
reformatted transformer_lens/loading_from_pretrained.py

All done! ✨ 🍰 ✨
1 file reformatted.

**Comment by mylesgoose** [comment]
ran test on 3.1 70b with this branch is working. And I ran that Black code black transformer_lens/loading_from_pretrained.py
reformatted transformer_lens/loading_from_pretrained.py

All done! ✨ 🍰 ✨
1 file reformatted. on the file that was causing the error

**Comment by mylesgoose** [comment]
> I'll probably get this in a release today

I got the vision model abliterated today. Using a guys script of hf. to convert the 70b abliterated model into the 90b model. So now just trying to go in reverse to extract the 70b model from 90b and abliterate it, then put back. It was funny because i turned on the wrong layer and the new vision model was refusing even the most benign requests.

**Comment by bryce13950** [comment]
I am sorry to say, but we cannot include the models from your hugging face account `mylesgoose`. LLaMA models are gated models. If someone wants to use LLaMA with TransformerLens, then they have to agree to Metas legal agreements on HuggingFace. If we include your fork, then someone could relatively easily get access to what are gated models, and that could open up TransformerLens to legal issues. 

There is a relatively easy way to use those models in TransformerLens without having them officially in the branch by providing the downloaded model, and using the official name. If you can remove them from your PR, then we can move forward with adding the rest of LLaMA 3.1.

**Comment by mylesgoose** [comment]
> I am sorry to say, but we cannot include the models from your hugging face account `mylesgoose`. LLaMA models are gated models. If someone wants to use LLaMA with TransformerLens, then they have to agree to Metas legal agreements on HuggingFace. If we include your fork, then someone could relatively easily get access to what are gated models, and that could open up TransformerLens to legal issues.
> 
> There is a relatively easy way to use those models in TransformerLens without having them officially in the branch by providing the downloaded model, and using the official name. If you can remove them from your PR, then we can move forward with adding the rest of LLaMA 3.1.
Meta has given permission to duplicate the gated models and distribute as long as it includes the licence file and the name Llama in the front of it. also also f the other models loaded there are also based on some form of other base model with a licence. meta is aware i duplicated the models as i had to amend them with he chat template for training. in a pr with them. it clearly says in the licence we are allowed to reproduce them.provided the naming convention and licence file is met.

**Comment by mylesgoose** [comment]
for example this model is also gated https://huggingface.co/meta-llama/Llama-2-7b-hf and yet you have three ungtaed equivalents here   "CodeLlama-7b-hf",
    "CodeLlama-7b-Python-hf",
    "CodeLlama-7b-Instruct-hf",

**Comment by mylesgoose** [comment]
1. License Rights and Redistribution.

    a. Grant of Rights. You are granted a non-exclusive, worldwide, 
non-transferable and royalty-free limited license under Meta’s intellectual property or other rights 
owned by Meta embodied in the Llama Materials to use, reproduce, distribute, copy, create derivative works 
of, and make modifications to the Llama Materials.  

    b. Redistribution and Use.  

        i. If you distribute or make available the Llama Materials (or any derivative works thereof), 
or a product or service (including another AI model) that contains any of them, you shall (A) provide
a copy of this Agreement with any such Llama Materials; and (B) prominently display “Built with Llama”
on a related website, user interface, blogpost, about page, or product documentation. If you use the
Llama Materials or any outputs or results of the Llama Materials to create, train, fine tune, or
otherwise improve an AI model, which is distributed or made available, you shall also include “Llama”
at the beginning of any such AI model name.

**Comment by bryce13950** [comment]
The models you are trying to include don't even exist. https://huggingface.co/mylesgoose/Llama-3.1-8B-Instruct. On top of that, a lot of the models that you have publicly available on your profile do not include the meta license in the repo. Meaning that, you are not redistributing them as per the license agreement you have just quoted. The CodeLlama models are transformative, and each model receives thousands of downloads a month. On top of that, they follow the license agreement that meta has outlined to a tee. 

We do not include private repositories of models. We do not include models that are seemingly not following the legal obligations they are obliged to follow. Not doing that could open up TransformerLens to legal consequences. You already have a way to load your fork into TransformerLens. Also, if you had opened a PR with them, and your changes have been merged into their fork, then what is special about your models? 

There is absolutely no need for you to put your personal fork into TransformerLens. If your fork is truly transformative, and all of a sudden you are getting hundreds of downloads a week, then we can discuss this again. For the time being, you need to remove these models, or else we are not going to be able to count your contribution.

**Comment by mylesgoose** [comment]
Hello I just checked every repo. And they all have the digital licence on the page.  Which is a link to metas licence. So I am obeying the contract with them. As per the missing models. They are still being pushed to the hub. It keeps crashing on uploads. Won't be long couple days maybe. Cheers

**Comment by bryce13950** [comment]
Well I hope you figure it out! When you get your models removed from this PR I will take another look. In the time being, someone else may add LLaMA 3.1 to TransformerLens, and if that happens before you get yours removed, I will just close this out.

**Review by bryce13950 — CHANGES_REQUESTED** [review]

**Comment by TacticalSpoon331** [comment]
Howdy ya'll

I'm not seeing support for 405b. Am I blind or have we not added it yet?

**Comment by bryce13950** [comment]
> Howdy ya'll
> 
> I'm not seeing support for 405b. Am I blind or have we not added it yet?


That model is around 800gb, so it is probably not likely to be supported officially due to the computing requirements needed just to load it. If it becomes requested more often, we may consider adding support for it, but to my knowledge, you are the first person to ask. I'll keep it in mind to see if there is more interest in using it.

File: transformer_lens/loading_from_pretrained.py [hunk 1]
```diff
@@ -151,10 +152,27 @@
...
+    "mylesgoose/Llama-3.1-405B-Instruct",
+    "mylesgoose/Llama-3.1-405B",
     "meta-llama/Llama-3.2-1B",
     "meta-llama/Llama-3.2-3B",
     "meta-llama/Llama-3.2-1B-Instruct",
     "meta-llama/Llama-3.2-3B-Instruct",
+    "mylesgoose/Llama-3.2-1B",
+    "mylesgoose/Llama-3.2-3B",
+    "mylesgoose/Llama-3.2-1B-Instruct",
+    "mylesgoose/Llama-3.2-3B-Instruct",
```
[the remark this answers is not in the document]
Comment (mylesgoose) [reply transformer_lens/loading_from_pretrained.py:175]: Well i would like to keep the models in this branch as that what im currently using. So i can keep on developing this branch with the support for the ungated ones as many people are downloading and using them.  are you able to pull it to your branch without deletion from my one?

**Comment by mylesgoose** [comment]
> Howdy ya'll
> 
> I'm not seeing support for 405b. Am I blind or have we not added it yet?

in this branch you can click above 405 b should be supported. as i set it to load its configuration form the config.json file. if you have enough ram. "meta-llama/Llama-3.1-405B-Instruct",

File: transformer_lens/loading_from_pretrained.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mylesgoose) [reply transformer_lens/loading_from_pretrained.py:175]: can you explain what actually is the issue with the models? they are not private models. they are all very popular on huggingface. they are allowing all of europe to access them. which is not currently possible due to eu regulations asking meta to make available the training data. having them there make it allot easier for me to continue development on the main branch of your repo. for example i have been abel now to abliterate the vison models and this was only possible by loading my own custom abliterated model with specific templates. It makes it allot more user friendly also a having the option for users. as they dont have to load the tokens each time. we are complying with the requirements of the agreement for the models. there exists a significant t amount of private or personal models in the lens repo now many which will never be used. there exists many gated models ungated. i personally cannot understand why you would want to remove a great feature? also you are not hosting the models you are simply enabling them to load.If i have neglected to follow a contract with meta i fail to see how it is your guys legal issue? the argument does not make sense. All of the negative issues you guys raised i have proven you where incorrect. so i fail to see how an extra 8 lines of code makes the repo unmergable for you guys.

**Comment by TacticalSpoon331** [comment]
Well Good Googly Moogly I just might have to give that a try

File: transformer_lens/loading_from_pretrained.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryce13950) [reply transformer_lens/loading_from_pretrained.py:175]: I am sorry, but we are not going to add them. The only two people in this project who have personal models in this project are Neel Nanda, and Arthur Conmy, the two people that started TransformerLens. Almost all of those models are under a gigabyte, which means it only takes a couple minutes to test their models. They are all also original models, and not forks of existing models, which means that they would not be able to be loaded in TransformerLens without being in the list of officially supported models.

It take about 2 hours to test a 70b model. You are trying to add 3 of them, which currently do not exist on huggingface to review. These 8 lines of code are going to add about 10-20 hours of maintenance time to this project a month. 

You also already can use these models in TransformerLens once LLaMA 3.1 is added by simply passing in `hf_model` with the official model id, which means that there is absolutely no reason for personal models forking LLaMA, or any other supported model to be added. 

You have yet to give any reason of what makes these models transformative vs the standard models. You brought up CodeLLaMA as an example of us supporting similar situations. They wrote a paper about why their models are transformative. If you would like to share any papers you have written that supports why these models are transformative, I am sure we would consider looking at that.

Also, I don't know what you are talking about in regards to this making these models available in the EU. I have had no issues accessing the official LLaMA models.

**Comment by mylesgoose** [comment]
https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct/discussions/22#66f482ecf14a7e752a0553c0

**Comment by TacticalSpoon331** [comment]
> https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct/discussions/28#66f52be3a3e770bd4eabdbbe they are blocked in Europe. The vision models. Because trained on copyright images. Apparently. Can you copy the pr to your repo and delete those line. Cheers.

Hey your fork works great for all of the default 3.1 models. Having trouble with quantized versions though... That being said I haven't tested quantized versions on the main fork but 🤷‍♂️ Have you tested quantized models of 3.1?

**Comment by mylesgoose** [comment]
> > https://huggingface.co/meta-llama/Llama-3.2-11B-Vision-Instruct/discussions/28#66f52be3a3e770bd4eabdbbe they are blocked in Europe. The vision models. Because trained on copyright images. Apparently. Can you copy the pr to your repo and delete those line. Cheers.
> 
> Hey your fork works great for all of the default 3.1 models. Having trouble with quantized versions though... That being said I haven't tested quantized versions on the main fork but 🤷‍♂️ Have you tested quantized models of 3.1?

You would have to edit the quantization settings in the config that's passed in your own model and load from local or install as editable install pip3 install -e . And then adjust the lmforcasual loading perhaps. On model config .py file

**Pull request closed without merging**